saleor/graphql/order/mutations/orders.py

```diff
@@ -824,6 +854,28 @@ def perform_mutation(cls, _root, info, id):
         delete_order_line(line_info)
         line.id = db_id
 
+        if order.is_shipping_required():
+            events.order_removed_products_event(
+                order=order,
+                user=info.context.user,
+                app=info.context.app,
+                order_lines=[(line.quantity, line)],
+            )
+            recalculate_order(order)
```

Comment:
Now, this piece of code is duplicated - we call it when shipping is required, but also when it's not, we call it at the end of the mutation. We could easily remove the duplication:
- invert the if condition to `if not order.is_shipping_required()`
- inside the `if` block - clear the shipping method (the logic that is done in lines 867-878)
- after exiting the `if` block, create the history event and recalculate order